### PR TITLE
Created Add Group and Delete Group helpers functions

### PR DESCRIPTION
**Pull request by majesticfish**
### Summary
- Two new functions in modules/groups/helpers.py
- helpers.create_group() creates a new group with specified attributes, also automatically creates a position associated with the group called "Member"
- helpers.delete_group() removes a group and all its positions
### Test Plan
- Two new test cases for these methods

### CODE REVIEW

File: donut/modules/groups/helpers.py [hunk 1]
```diff
@@ -237,12 +237,63 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+    position holders entries and positions
+
+    Arguments:
+        group_id: id of the group to be deleted
+    '''
+    group_positions = get_group_positions(group_id)
+    for pos in group_positions:
+        delete_position(pos["pos_id"])
+    s = "DELETE FROM groups WHERE group_id=%d"
+    s = s % group_id
```
Comment (davistardif) [inline donut/modules/groups/helpers.py]: Why do this here instead of at the `cursor.execute()` call?

File: donut/modules/groups/helpers.py [hunk 2]
```diff
@@ -237,12 +237,63 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+        "members_can_send", "visible", "admin_control_members"
+    ]
+    query = "INSERT INTO groups (" + ','.join(fields) + ")"\
+        "VALUES (%s, %s, %s, %r, %r, %r, %r, %r)"
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (group_name, group_desc, group_type, newsgroups,
+                               anyone_can_send, members_can_send, visible,
+                               admin_control_members))
+        add_position(cursor.lastrowid, "Member")
+
```
Comment (davistardif) [inline donut/modules/groups/helpers.py:277]: Would be nice to return the group_id for the group that just got created. For example, you could use the returned group_id in place of a constant in your tests, so if someone adds a new group to the test data it doesn't break your tests

**Review by davistardif — COMMENTED** [review]
Some small suggestions. Other than that, looks good to me

File: donut/modules/groups/helpers.py [hunk 3]
```diff
@@ -237,12 +237,63 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+        memberse_can_send: Toggles if members can send emails to this group
+        visible: Toggles if the group is visible
+        admin_control_members: toggles if administrator can control membership
+    """
+    fields = [
+        "group_name", "group_desc", "type", "newsgroups", "anyone_can_send",
+        "members_can_send", "visible", "admin_control_members"
+    ]
+    query = "INSERT INTO groups (" + ','.join(fields) + ")"\
+        "VALUES (%s, %s, %s, %r, %r, %r, %r, %r)"
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Is there a reason why these are not all `%s`?

File: donut/modules/groups/helpers.py [hunk 4]
```diff
@@ -237,12 +237,63 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+    '''
+    Deletes the group specified with the group_id and all assocaited
+    position holders entries and positions
+
+    Arguments:
+        group_id: id of the group to be deleted
+    '''
+    group_positions = get_group_positions(group_id)
+    for pos in group_positions:
+        delete_position(pos["pos_id"])
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Couldn't this be accomplished with an `ON DELETE CASCADE` in the `positions` and `position_holders` tables?

Comment (calebsander) [inline donut/modules/groups/helpers.py]: Couldn't this be accomplished in SQL instead with an `ON DELETE CASCADE`?

File: donut/modules/groups/helpers.py (hunk 3 above)
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Why are these not all `%s`?

File: donut/modules/groups/helpers.py [hunk 5]
```diff
@@ -237,12 +237,63 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+        "group_name", "group_desc", "type", "newsgroups", "anyone_can_send",
+        "members_can_send", "visible", "admin_control_members"
+    ]
+    query = "INSERT INTO groups (" + ','.join(fields) + ")"\
+        "VALUES (%s, %s, %s, %r, %r, %r, %r, %r)"
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (group_name, group_desc, group_type, newsgroups,
+                               anyone_can_send, members_can_send, visible,
+                               admin_control_members))
+        add_position(cursor.lastrowid, "Member")
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Put `cursor.lastrowid` in a local variable and call `add_position()` from outside the `with` block. No need to hold multiple cursors at the same time.

**Review by calebsander — CHANGES_REQUESTED** [review]

File: donut/modules/groups/helpers.py (hunk 3 above)
Comment (majesticfish) [reply donut/modules/groups/helpers.py]: They are boolean values, does the formatting still work if it is `%s`?

Comment (calebsander) [reply donut/modules/groups/helpers.py]: I would imagine so; `'%s' % True` and `'%r' % True` give the same value

File: donut/modules/groups/helpers.py [hunk 6]
```diff
@@ -237,12 +237,61 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+        visible: Toggles if the group is visible
+        admin_control_members: toggles if administrator can control membership
+    """
+    fields = [
+        "group_name", "group_desc", "type", "newsgroups", "anyone_can_send",
+        "members_can_send", "visible", "admin_control_members"
+    ]
+    query = "INSERT INTO groups (" + ','.join(fields) + ")"\
+        "VALUES (%s, %s, %s, %s, %s, %s, %s, %s)"
+    new_group_id = -1
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Why initialize this value if it's always overwritten?

File: donut/modules/groups/helpers.py [hunk 7]
```diff
@@ -237,12 +237,61 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+        anyone_can_send: Toggles if anyone can send emails to this group
+        memberse_can_send: Toggles if members can send emails to this group
+        visible: Toggles if the group is visible
+        admin_control_members: toggles if administrator can control membership
+    """
+    fields = [
+        "group_name", "group_desc", "type", "newsgroups", "anyone_can_send",
+        "members_can_send", "visible", "admin_control_members"
+    ]
+    query = "INSERT INTO groups (" + ','.join(fields) + ")"\
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: If `fields` are fixed, might be cleaner to just hard-code them into `query`

File: tests/modules/groups/test_groups.py [hunk 8]
```diff
@@ -159,6 +159,34 @@ def test_create_pos_holders(client):
...
+        "members_can_send", "newsgroups", "visible", "admin_control_members"
+    ]) == {
+        "group_name": "Page House",
+        "group_desc": "May the work be light",
+        "type": "House",
+        "newsgroups": 0,
+        "anyone_can_send": 0,
+        "members_can_send": 1,
+        "visible": 1,
+        "admin_control_members": 1,
```
Comment (calebsander) [inline tests/modules/groups/test_groups.py]: Seems a little counter-intuitive that these values are booleans when passed to `create_group()` but integers in the result of `get_group_data()`. Would be nice to convert them back to booleans.

File: donut/modules/groups/helpers.py [hunk 9]
```diff
@@ -237,12 +237,61 @@ def create_position_holder(pos_id, user_id, start_date, end_date):
...
+    '''
+    Deletes the group specified with the group_id and all assocaited
+    position holders entries and positions
+
+    Arguments:
+        group_id: id of the group to be deleted
+    '''
+    s = "DELETE FROM groups WHERE group_id=%s"
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(s, (group_id))
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Parens unnecessary

**Review by calebsander — APPROVED** [review]

**Pull request merged**